### PR TITLE
Pause state

### PR DESCRIPTION
**Pull request by katherineperks**
Pause is no longer synced immediately, but on the next cycle. Pausing at any point during the task ends the current trial and resuming sends bmi3d to the "wait" state of the next trial. The interrupted trial will not have a corresponding "TRIAL_END" event because it did not reach an end state (i.e. penalty or reward). Instead, "PAUSE_START" is sent upon pausing and "PAUSE_END" upon resuming. 
To find the number of trials played, both "TRIAL_END" and "PAUSE_START" (or "PAUSE_END") events must be counted.

### CODE REVIEW

**Review by leoscholl — APPROVED** [review]
Looks good. Only problem i have is that pause will be broken for all the other tasks as soon as we merge this PR! Don't know if there's any way to get around that other than making a pause state for target capture task now.

**Comment by katherineperks** [comment]
I added a check in experiment.py to use the old method of sending a PAUSE sync event immediately if the current experiment has no pause state. Otherwise, pausing will send a PAUSE_START sync event and resuming will send a PAUSE_END sync event.

**Pull request merged**
